dfastmi/gui/dialog_view.py

```diff
@@ -102,37 +105,49 @@ def update_reach(self, data):
         self.slength.setText(self.view_model.slength)
     
     def update_condition_files(self):
-        for condition_tab_index, reference_file in enumerate(self.view_model.reference_files):
-            tab = self._tabs.widget(condition_tab_index +1)
-            if tab:
-                prefix = str(condition_tab_index)+"_"
-                key = prefix +"file1"
-                input_textbox = tab.findChild(QtWidgets.QLineEdit, key)
-                if input_textbox:
-                    input_textbox.setText(reference_file)
+        for condition_discharge, reference_file in self.view_model.reference_files.items():
+            prefix = str(condition_discharge)+"_"
+            key = prefix +"file1"
+            input_textbox = self.general_widget.findChild(QtWidgets.QLineEdit, key)
+            if input_textbox:
+                input_textbox.setText(reference_file)
         
-        for condition_tab_index, measure_file in enumerate(self.view_model.measure_files):
-            tab = self._tabs.widget(condition_tab_index +1)
-            if tab:
-                prefix = str(condition_tab_index)+"_"
-                key = prefix +"file2"
-                input_textbox = tab.findChild(QtWidgets.QLineEdit, key)
-                if input_textbox:
-                    input_textbox.setText(measure_file)
+        for condition_discharge, measure_file in self.view_model.measure_files.items():        
+            prefix = str(condition_discharge)+"_"
```

Comment:
These two for-loops can easily be deduplicated. 

```
    def update_condition_files(self):
        for condition_discharge, reference_file in self.view_model.reference_files.items():
            self._update_condition_file_field("file1", condition_discharge, reference_file)
        
        for condition_discharge, measure_file in self.view_model.measure_files.items():        
            self._update_condition_file_field("file2", condition_discharge, measure_file)
    
    def _update_condition_file_field(self, field_postfix: str, condition_discharge: float, condition_file: str) -> None:
            prefix = str(condition_discharge) + "_"
            key = prefix + field_postfix
            input_textbox = self.general_widget.findChild(QtWidgets.QLineEdit, key)
            if input_textbox:
                input_textbox.setText(condition_file)
```

Reply:
good idea, updated